channel.go

```diff
@@ -758,6 +760,16 @@ func (ch *Channel) connectionCloseStateChange(c *Connection) {
 
 func (ch *Channel) onClosed() {
 	removeClosedChannel(ch)
+
+	ch.mutable.Lock()
+	select {
+	case <-ch.mutable.closed:
```

Comment:
Do we need this check, I think `onClosed` is guaranteed to only happen once right?

Reply:
I thought so too, but it looks like it's called from both `Close` and `connectionCloseStateChange`.

And the tests appear to fail if I don't guard it.